### PR TITLE
Read 4 byte integers and floats and handle more unknown bytecode

### PR DESCRIPTION
**Pull request by Quackster**
Just an attempt to handle more bytecode and remove some of the "UNK_xx" found when viewing in lscrtoscript.

For example, when uploading hh_photo.cst

```
on countCS(me, tImg)
  tL = [3, 2, 73, 28, 83, 21, 43, 90, 92, 91, 37, 4, 3, 84, 12, 102, 103, 108, 97, 43, 44, 89, 109, 65, 61, -4, 76]
  tA = 0
  tW = tImg.width
  tH = tImg.height
  i = 1
  repeat while i <= 100
    tA = ERROR mod ERROR mod tA + tImg.getPixel(i mod tW, i * i mod tH).paletteIndex * tL.getAt(i mod tL.count + 1).undefined
    i = 1 + i
  end repeat
  return(tA)
  exit
end
```

Now turns into

```
on countCS(me,tImg)
  tL = [3, 2, 73, 28, 83, 21, 43, 90, 92, 91, 37, 4, 3, 84, 12, 102, 103, 108, 97, 43, 44, 89, 109, 65, 61, -4, 76]
  tA = 0
  tW = tImg.width
  tH = tImg.height
  i = 1
  repeat while i <= 100
    tA = tA + tImg.getPixel(i mod tW, i * i mod tH).paletteIndex * tL.getAt(i mod tL.count + 1) mod 85000
    i = 1 + i
  end repeat
  return(tA)
  exit
end
```

### CODE REVIEW

**Comment by hqkirkland** [comment]
Awesome progress!

**Comment by Brian151** [comment]
i'll review this when i'm not so tired...
however, this repo is technically inactive

https://github.com/Earthquake-Project
is the current place to find everything

**Comment by Quackster** [comment]
I pushed it before because I couldn't find lscrtoscript project on the Earthquake-Project 😛

**Comment by hqkirkland** [comment]
> I pushed it before because I couldn't find lscrtoscript project on the Earthquake-Project 😛

I believe this is the new repo:

https://github.com/Earthquake-Project/ProjectorRays

**Comment by Brian151** [comment]
you should indeed PR projectorrays

tbh, it is unfortunate I am the one to have to try answering this question, as the bytecode still largely confuses me. it doesn't help, either there's still a bunch of those unknowns... 

my personal opinion is to be careful with this... 
at least, by our earlier observations, the highest length for a bytecode has thus far been four bytes, this would mean there are five byte codes. Of course, anything is possible with this format.

All I ever really did with bytecode is resolve some ambiguity/inaccuracy from previous efforts made by other people, and assign names to all the opcodes. well, all the known ones. also note the 'special' opcodes (all getters/setters), which are context-sensitive in what a previous opcode had done. 

you're only showing the output lingo script, and not the disassembly, so it's hard to follow your example, anyways... also, is this a cst obtained by unprotecting a cct/cxt? or do you have the original source code?

anyways, you'd be much better off to re-submit this to the typescript port of projectorrays

**Comment by Quackster** [comment]
>you're only showing the output lingo script, and not the disassembly, so it's hard to follow your example, anyways... also, is this a cst obtained by unprotecting a cct/cxt? or do you have the original source code?

It was an unprotected .cct who used 4 byte integers, demonstrating how fixing the bytecode reading is beneficial to recovering workable Lingo script.

>anyways, you'd be much better off to re-submit this to the typescript port of projectorrays

I know, but you had no repo for lscrotscript on the new project page.

**Comment by Brian151** [comment]
seeing the actual example file would be far more beneficial to improving the decompiler. we're otherwise flying blind and don't really know which way to turn next if this is wrong. 


https://github.com/Earthquake-Project/ProjectorRays
it's right there

lscrtoscript in its original form, is abandoned
